backstory/src/main/java/com/google/sps/images/data/AnnotatedImage.java

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.images.data;
+
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.lang.IllegalArgumentException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public final class AnnotatedImage {
+  private final byte[] rawImageData;
+  private final List<EntityAnnotation> labelAnnotations;
+
+  /**
+   * AnnotatedImage is an immutable type which represents an image annotated with labels and other
+   * related analytics.
+   *
+   * The image is represented as the byte data in the rawImageData byte array, and the annotated
+   * analytics are represented as labelAnnotations, a list of Vision EntityAnnotation objects, which
+   * each represent individual labels.
+   *
+   * Abstraction Function(rawImageData, labelAnnotations) =
+   * An image represented by the bytes in rawImageData, with a list of labels annotated
+   * to the image, represented by the EntityAnnotation(s) in labelAnnotations.
+   *
+   * Representation Invariant:
+   * rawImageData will always represent the same image, it is immutable. It must be non-null. It
+   * must be non-empty. Once set, labelAnnotations will never change, it is immutable. It must be
+   * non-null.
+   *
+   * Safety from Representation Exposure:
+   * Representation fields are only returned as immutable representations, or (in the case of
+   * rawImageData) as a defensive copy.
+   */
```

Comment:
Make this JavaDoc?

```
/**
 * Class documentation
 */
class ClassName {
...
```

Implementation comments are harder to discover--people reading code don't often navigate to the class unless something's unclear; often IDE's allow you to see documentation just by hovering over a documented thing.

Reply:
Sounds good. I will make this javadoc from now on.